pkg/util/strategicpatch/patch_test.go

```diff
@@ -0,0 +1,433 @@
+        - name: 2
+    patch:
+      mergingList:
+        - name: 1
+          mergingList:
+            - name: 1
+              value: 1
+    result:
+      mergingList:
+        - name: 1
+          mergingList:
+            - name: 1
+              value: 1
+            - name: 2
+              value: 2
+        - name: 2
+  - description: update map when map should be replaced
+    original:
+      name: 1
+      value: 1
+    patch:
+      value: 1
+      __patch: strict
+    result:
+      value: 1
+  - description: merge empty merge lists
+    original:
+      mergingList: []
+    patch:
+      mergingList: []
+    result:
+      mergingList: []
+  - description: delete others in a map
+    original:
+      name: 1
+      value: 1
+    patch:
+      __patch: strict
+    result: {}
+  - description: delete item from a merge list
+    original:
+      mergingList:
+        - name: 1
+        - name: 2
+    patch:
+      mergingList:
+        - __patch: delete
+          name: 1
+    result:
+      mergingList:
+        - name: 2
+  - description: add and delete item from a merge list
+    original:
+      merginglist:
+        - name: 1
+        - name: 2
+    patch:
+      merginglist:
+        - name: 3
+        - __patch: delete
+          name: 1
+    result:
+      merginglist:
+        - name: 2
+        - name: 3
+  - description: delete all items from a merge list
+    original:
+      mergingList:
+        - name: 1
+        - name: 2
+    patch:
+      mergingList:
+        - __patch: strict
+    result:
+      mergingList: []
+sortMergeListTestCases:
+  - description: sort one list of maps
+    original:
+      mergingList:
+        - name: 1
+        - name: 3
+        - name: 2
+    sorted:
+      mergingList:
+        - name: 1
+        - name: 2
+        - name: 3
+  - description: sort lists of maps but not nested lists of maps
+    original:
+      mergingList:
+        - name: 2
+          nonMergingList:
+            - name: 1
+            - name: 3
+            - name: 2
+        - name: 1
+          nonMergingList:
+            - name: 2
+            - name: 1
+    sorted:
+      mergingList:
+        - name: 1
+          nonMergingList:
+            - name: 2
+            - name: 1
+        - name: 2
+          nonMergingList:
+            - name: 1
+            - name: 3
+            - name: 2
+  - description: sort lists of maps and nested lists of maps
+    fieldTypes:
+    original:
+      mergingList:
+        - name: 2
+          mergingList:
+            - name: 1
+            - name: 3
+            - name: 2
+        - name: 1
+          mergingList:
+            - name: 2
+            - name: 1
+    sorted:
+      mergingList:
+        - name: 1
+          mergingList:
+            - name: 1
+            - name: 2
+        - name: 2
+          mergingList:
+            - name: 1
+            - name: 2
+            - name: 3
+  - description: merging list nested in a non merging list should NOT sort
```

Comment:
I guess this is actually saying, merging list should not merge when nested in a non merging list, and therefore it shouldn't sort either. That's because a replacing list is going to replace wholesale - it's not going to recurse into the list to find lists to merge, because it wouldn't make any sense because the list elements of the non-merging list can't be matched up in the first place.